Query role

odeke-em opened the pull request:
Fixes #200 or at least implements the last of the requested handlers/rest methods in that issue.
Depends on #208 which after merge into unstable will make this PR comprised of just one commit.

odeke-em commented:
/cc @ethanfrey

ebuchman commented on modules/coin/rest/handlers.go:
```diff
@@ -118,6 +121,25 @@ func doSend(w http.ResponseWriter, r *http.Request) {
 	common.WriteSuccess(w, tx)
 }
 
+type RoleInput struct {
```
should this struct be duplicated in modules/coin and modules/roles ?

odeke-em replied:
Thanks for pointing that out. So this PR is based off #208 and since it was taking long to get it reviewed I based this branch of prior versions of it and made this one, with RoleInput still in modules/coin/rest instead of modules/roles/rest. Once #208 is merged, I'll rebase it here and it'll be gone :)

ebuchman commented on modules/roles/rest/handlers.go:
```diff
@@ -0,0 +1,107 @@
+	common.WriteSuccess(w, recvRole)
+}
+
+func RegisterQueryRole(r *mux.Router) error {
```
Can we get a comment on this public function

odeke-em replied:
Thanks for the catch, I'll do so.

odeke-em commented:
Addressed your comments @ebuchman, however as spoken with and examined with @ethanfrey offline, some is up when we try to do `node.ABCIQuery("/key", key, true)` so this can hold off until renaming is done.

odeke-em commented:
@ethanfrey could we add more context as to why we close PRs while doing them? I ask for this so that in the future when revisiting them, we can know what's up and why we closed them.

ethanfrey commented:
Sorry for the confusion.  I think this happened during the rename. 

You are more than welcome to rebase and update all imports to Sdk and reopen it.

ethanfrey commented:
Oh, yeah, I deleted the unstable branch in this repo after merging it into develop. That would have auto-closed it.
